BEIMA.Client/src/pages/Users/AddUserPage.js

```diff
@@ -0,0 +1,86 @@
+import styles from './AddUserPage.module.css';
+import { useOutletContext, Link } from 'react-router-dom';
+import { IoArrowBack } from "react-icons/io5";
+import { Card, Button, Form, Row, Col } from 'react-bootstrap';
+import FormListWithErrorFeedback from '../../shared/FormList/FormListWithErrorFeedback.js';
+import { useEffect, useState } from "react";
+
+const AddUserPage = () => {
+  const mandatoryUserFields = {
+    "Username" : "",
+    "First Name" : "",
+    "Last Name" : "",
+    "Role" : ""
+  }
+  
+  const [userFields] = useState(mandatoryUserFields);
+  const [errors, setErrors] = useState(mandatoryUserFields);
+  const [setPageName] = useOutletContext();
+  const [fullUserJSON, setFullUserJSON] = useState({});
+  
+  useEffect(() => {
+    setPageName('Add User')
+  }, [setPageName])
+  
+  // gathers all the input and puts it into JSON
+  function createJSON(addButtonEvent){
+    let formFields = addButtonEvent.target.form.elements;
+    let fieldValues = {};
+    let newErrors = {};
+
+    for(let i = 0; i < formFields.length; i++){
+      let formName = formFields[i].name;
+      let fieldNames = Object.keys(userFields);
+      
+      if(fieldNames.includes(formName)){
+        let formJSON =  {[formName] : formFields[i].value};
+        
+        Object.assign(fieldValues, formJSON);
+      }
+    }
+
+    setFullUserJSON(fieldValues);
+    
+    // replace with user api calls when ready
+    console.log(fullUserJSON);
+
+    if ( Object.keys(newErrors).length > 0 ) {
```

Comment:
I can't think of any errors they could encounter when filling out this information. Maybe it's not needed on this form?

Reply:
The only thing I could think of is the Role remaining a text field rather than being converted to a dropdown when API integration happens for the page to guard against mistyping a role name. Other than that I agree, and I think changing it to a dropdown is the way to go once we have role admin in place as well.
I think it's safe to remove.

- [ ] remove error check state var/if block